### ISSUE TITLE
Element detached from the DOM error about 0.1% of the time with Angular

### ISSUE THREAD

**Issue by pawel02** [cypress-io/cypress#21058]
### Current behavior

I have the following CustomGet for fetching elements that has been outlined in issue https://github.com/cypress-io/cypress/issues/7306
```
Cypress.Commands.add('CustomGet', (selector: string, options?: any) => {
    cy.waitUntil(() => 
        cy.get(selector, options)
        .wait(10)
        .then($el => Cypress.dom.isAttached($el)),
        {timeout:20000, interval: 500});
    cy.get(selector, options).first().scrollIntoView();
    return  cy.get(selector, options);
});
```
However, whenever I run the tests both locally and on the build node about 0.1% of the time I will still get an error saying that the Element is detached from the DOM.
My front-end is written using Angular and it utilises WebSockets. I can get updates from my backend at any point causing the front-end to update and I believe that is causing the issue as there is a small possibility of the reference to the element becoming invalid in that split second of returning the reference back. 

### Desired behavior

I would like to be able to simply do a cy.get on an element and if the reference becomes invalid for cypress to imminently try to re-query the element. Not really sure if this should be part of the library or whether I should just write some sort of a wrapper for this. 

### Test code to reproduce

Unfortunately I am not allowed to share the source code as its closed source however, if needed I can try to create an example repo with angular and cypress.

### Cypress Version

9.5.1

**Comment by davidmunechika** [comment]
Thanks for opening an issue! I understand that you can't share the source code, but it would be great if you could try and create an example repo that reproduces the issue. That will allow us to move forward with investigating it and finding a solution.

**Comment by jennifer-shehane** [comment]
Duplicate of https://github.com/cypress-io/cypress/issues/7306 We are planning to put some time into investigating this issue this quarter.

**Comment by allen-beamry** [comment]
@jennifer-shehane May I know the progress of investigating this issue?
I upgraded to Cypress v10+, and started seen flaky tests due to the `Element detached from the DOM`.

> Duplicate of #7306 We are planning to put some time into investigating this issue this quarter.

**Comment by suppadeliux** [comment]
Having the same issue! Before upgrading my project to Angular 15 and Node 18, I was able to run my tests successfully for months everyday, but after the update, things radically started to be very flaky, and having detached DOM elements, inputs animating errors, and click animating as well!

The weird thing is that my tests didn't change between the updates.

Updated my project to cypress 12, and things didn't improve.

It's very frustrating, but i guess, having websockets, rxjs, angular are tricky when testing with cypress.

I hope this is updated some day!
